lib/oli_web/controllers/api/scheduling_controller.ex

```diff
@@ -171,6 +171,23 @@ defmodule OliWeb.Api.SchedulingController do
     end
   end
 
+  def clear(conn, %{"section_slug" => _section_slug}) do
+    section = conn.assigns.section
+
+    if can_access_section?(conn, section) do
+      case Scheduling.clear(section) do
+        {changes_count, nil} when is_integer(changes_count) ->
+          conn
+          |> json(%{"result" => "success"})
+
+        error ->
```

Comment:
I agree, but.. what scenario do you think should handle some kind of error that occurs when doing `Repo.Update_all`?

Reply:
According to the docs, [update_all](https://hexdocs.pm/ecto/Ecto.Repo.html#c:update_all/3) will never return an error, even if it fails. I guess one way of knowing whether an insert failed or not is pattern matching on `{0, _}`, since 0 records inserted could mean that the operation failed, but we are not 100% about that.